local/src/core/ProxyVerifier.cc

```diff
@@ -456,11 +463,22 @@ swoc::Errata Session::run_transactions(const std::list<Txn> &txn_list,
 
   for (auto const &txn : txn_list) {
     if (this->is_closed()) {
-      this->do_connect(real_target);
+      errata.note(this->do_connect(real_target));
+      if (!errata.is_ok()) {
+        errata.error(R"(Failed to reconnect HTTP/1 key={}.)", txn._req.make_key());
+      }
     }
+    const auto before = GetUTimestamp();
     errata.note(this->run_transaction(txn));
+    const auto after = GetUTimestamp();
     if (!errata.is_ok()) {
-      errata.error(R"(Failed url={}.)", txn._req._url);
+      errata.error(R"(Failed key={}.)", txn._req.make_key());
+    }
+
+    const auto elapsed_us = after - before;
+    if ((elapsed_us / 1000000) > 2) {
```

Comment:
No! Use `std::chrono` and cast to the time scale you want. E.g. `static_cast<std::chrono::seconds>(elapsed_us) > 2`. This is a key point of the library, to avoid having large arbitrary constants you hope you get right.